### PR TITLE
[METAL][TR][AMR] AMR and Text Extractions Linking

### PR DESCRIPTION
**Pull request by enoriega**
This branch contains the AMR <-> TR linking and removes the legacy linker from the source tree

### CODE REVIEW

**Review by myedibleenso — CHANGES_REQUESTED** [review]
Library fails to install.  Try installing the extras separately to debug (ex. `pip install .[tr]`).

File: pyproject.toml [hunk 1]
```diff
@@ -71,7 +83,8 @@ all = ["skema[core]", "skema[dev]", "skema[doc]", "skema[demo]"]
 "skema.program_analysis" = "skema/program_analysis"
 "skema.skema_py" = "skema/skema_py"
 # re-map skema/text_reading/python to skema.text_reading
-"skema.text_reading.mention_linking" = "skema/text_reading/mention_linking"
+"skema.text_reading" = "skema/text_reading/python"
+"skema.metal.model_linker" = "skema/metal/model_linker/skema_model_linker"
```
[the remark this answers is not in the document]
Comment (myedibleenso) [reply pyproject.toml:86]: Including the data directory as a module probably isn't what we want in this case (there is Python code in there).  I believe you'll need to alter a couple of different entries in the pyproject.toml:

[tool.setuptools.packages.find]
where = ["data"]

[tool.setuptools.package-data]
# the extensions you need
data = ["*.py", "*.f95"]

**Comment by myedibleenso** [comment]
In the "Create documentation for Python components (API docs)" step of the "Unit tests and project documentation" workflow, we generate Python API docs from type annotations and docstrings. At least one of your CI errors relates to this process.
- If `skema.text_reading.mention_linking` has been removed, you should remove this line: https://github.com/ml4ai/skema/blob/24303e1c1d93de3716d7b1f1ee3d4793994a2733/.github/workflows/tests-and-docs.yml#L72
- If it's only altered (new name/path), adjust the path and match it the corresponding entry in mkdocs.yml: https://github.com/ml4ai/skema/blob/24303e1c1d93de3716d7b1f1ee3d4793994a2733/mkdocs.yml#L51

File: skema/metal/model_linker/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+Link ASKEM models, such as `AMR` and `GrometFN` with text extractions in their [canonical form](https://github.com/ml4ai/ASKEM-TA1-DataModel).
+Currently, we support linking only to `AMR` representations, such as _PetriNets_ and _RegNets_.
+Soon, we will support linking to Gromet function networks.
+
+## Installation
+
+Clone this repository, and run:
+```shell
+pip install .
```
Comment (myedibleenso) [inline skema/metal/model_linker/README.md:11]: Does this work if installed separately from the top-level pyproject.toml-based build?

Comment (enoriega) [reply skema/metal/model_linker/README.md:11]: No. It was from before integrating it to the uber python package.
This goes away

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (enoriega) [reply pyproject.toml:86]: If you're referring to `skema/data/`, that isn't mine, but we can certainly tuck it into this PR

[the remark this answers is not in the document]
Comment (myedibleenso) [reply pyproject.toml:86]: Ah, sorry.  I think I got the line numbers mixed up (as well as the referenced data dir).  I believe I added that.  The inner data dir only contains an image meant to be used in conjunction with img2mml.  Let's leave it for now (this PR is complicated enough).

We can figure out a better way to include data for all submodules later.

**Review by myedibleenso — APPROVED** [review]

**Pull request merged**